### PR TITLE
docs: add headers to files, more debug statements and instructions how to debug

### PR DESCRIPTION
**Pull request by jayu**
## Motivation

Since codebase looks mysterious for the newcomers, I added headers containing high-level descriptions of file purpose and added more debug statements to easier analyze how Linaria works.

## Summary

- added debug statements
- added comment headers

## Test plan

Read comments and check for logic and grammar mistakes.

### CODE REVIEW

File: src/babel/utils/logger.ts [hunk 1]
```diff
@@ -60,3 +60,9 @@ export const debug = log.bind(null, 'debug');
 export const info = log.bind(null, 'info');
 export const warn = log.bind(null, 'warn');
 export const error = log.bind(null, 'error');
+
+export const debugNewLine = () => {
+  if (process.env.DEBUG === 'linaria*') {
```
Comment (Anber) [inline src/babel/utils/logger.ts:65]: It won't work for cases like `DEBUG=linaria:lazy-deps:*`

File: src/babel/extract.ts [hunk 2]
```diff
@@ -93,14 +104,15 @@ export default function extract(_babel: any, options: StrictOptions) {
           state.index = -1;
           state.dependencies = [];
           state.replacements = [];
+          debugNewLine();
+          debug('start extraction for:', state.file.opts.filename);
```
Comment (Anber) [inline src/babel/extract.ts:108]: The first arg of `debug` should be a namespace which can be used for logs filtering.

File: src/babel/utils/logger.ts (hunk 1 above)
Comment (jayu) [reply src/babel/utils/logger.ts:65]: Good point, I will fix it

Comment (jayu) [reply src/babel/utils/logger.ts:65]: I decided to remove it at all, I initially used it to split logs for particular files, but it seems to not be reasonable to debug more than one file at once.

File: src/babel/extract.ts (hunk 2 above)
Comment (jayu) [reply src/babel/extract.ts:108]: fixed

**Pull request merged**